test/spark.js

```diff
@@ -363,3 +363,33 @@ test('calculateDelayBeforeNextTask() handles one task per round', () => {
   })
   assertEquals(delay, 60_000)
 })
+
+  // Should abort after hitting maxFetchDurationMs and set requestIdleTimeout = true
+test('fetchCAR triggers timeout after long retrieval', async () => {
+  const sleep = (ms) => new Promise(resolve => setTimeout(resolve, ms))
+  const fetch = async (_url, { signal }) => {
+    return {
+      status: 200,
+      ok: true,
+      body: (async function * () {
+        while (true) {
+          if (signal.aborted) {
+            throw new DOMException('Aborted', 'AbortError')
+          }
+          yield new Uint8Array([0])
+          await sleep(500)
+        }
+      })()
+    }
+  }
+
+  const spark = new Spark({ fetch })
+  const stats = newStats()
+
+  await spark.fetchCAR('http', '/dns/example.com/tcp/80/http', KNOWN_CID, stats, {
+    maxFetchDurationMs: 1000
```

Comment:
```suggestion
    maxFetchDurationMs: 0
```

This way the test finishes sooner